File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        minPrice = prices[0]
        maxProfit = 0

        for i in range(len(prices)):
            if prices[i] < minPrice:
                minPrice = prices[i] #buying

            profit = prices[i] - minPrice #finding profit

            if profit > maxProfit:
                maxProfit = profit

        return maxProfit

        

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minPrice = prices[0];
        int maxProfit = 0;

        for(int i = 0;i<prices.size();i++){

            // checking the lowest buy price and saving in the minPrice Variable
            if(prices[i]<minPrice){
                minPrice = prices[i];
            }

            // finding the difference(profit) from Buy price and selling price
            int profit = prices[i] - minPrice;

            // if find greatest profit then change the maxProfit 
            if(profit > maxProfit){
                maxProfit = profit;
            }


        }
        return maxProfit;
        
    }
};